libafl/src/inputs/mod.rs

```diff
@@ -117,15 +117,112 @@ impl Input for NopInput {
 }
 impl HasTargetBytes for NopInput {
     fn target_bytes(&self) -> OwnedSlice<u8> {
-        OwnedSlice::from(vec![0])
+        OwnedSlice::from(vec![])
+    }
+}
+
+impl HasLen for NopInput {
+    fn len(&self) -> usize {
+        0
+    }
+}
+
+/// Target bytes wrapper keeping track of the current read position.
+/// Convenient wrapper when bytes must be split it in multiple subinputs.
+#[derive(Debug)]
+pub struct BytesReader<'a, I>
```

Comment:
I feel like std library functions for this use case exist?

Reply:
Look at cursor, maybe?

Reply:
https://doc.rust-lang.org/std/io/struct.Cursor.html

Reply:
Yes, my first idea was actually to use `Cursor`, but it does not really offer an interface to extract a slice without any copy while moving the cursor forward, hence this implem. It has some good side effects, like the fine control on what to do when reaching the end of the reader and reuse `OwnedSlice`, which keeps things consistent with libafl api.